test/test_prune_restarts.py

```diff
@@ -0,0 +1,223 @@
+        print("setup_module      module:%s" % module.__name__)
+
+    # Should be taken care of by teardown, in case remnants lying around
+    try:
+        shutil.rmtree(tmpdir)
+    except FileNotFoundError:
+        pass
+
+    try:
+        tmpdir.mkdir()
+        labdir.mkdir()
+        ctrldir.mkdir()
+        make_all_files()
+    except Exception as e:
+        print(e)
+
+
+def teardown_module(module):
+    """
+    Put any test-wide teardown code in here, e.g. removing test outputs
+    """
+    if verbose:
+        print("teardown_module   module:%s" % module.__name__)
+
+    try:
+        shutil.rmtree(tmpdir)
+        print('removing tmp')
+    except Exception as e:
+        print(e)
+
+
+@pytest.fixture(autouse=True)
+def teardown():
+    # Run test
+    yield
+
+    # Remove any created restart files
+    remove_expt_archive_dirs(type='restart')
+
+
+def create_test_2Y_1_month_frequency_restarts():
+    """Create 2 years + 1 month worth of mom restarts directories
+    with 1 month runtimes - starting from 1900/02/01 to 1902/02/01
+    e.g (run_date, restart_directory)
+    (1900/02/01, restart000)
+    (1900/03/01, restart001)
+     ...
+    (1902/02/01, restart024)"""
+    restart_dts = []
+    for year in [1900, 1901, 1902]:
+        for month in range(1, 13):
+            if (year == 1900 and month == 1) or (year == 1902 and month > 2):
+                # Ignore the first date and dates from 1902/03/01 onwards
+                continue
+            restart_dts.append(f"{year}-{month}-01 00:00:00")
+
+    for index, run_dt in enumerate(restart_dts):
+        make_ocean_restart_dir(start_dt="1900-01-01 00:00:00",
+                               run_dt=run_dt,
+                               calendar=4,
+                               restart_index=index,
+                               additional_path='ocean')
+
+
+def write_test_config(restart_freq, restart_history=None):
+    test_config = copy.deepcopy(config)
+    test_config['model'] = 'access-om2'
+    test_config['submodels'] = [
+        {'name': 'atmosphere', 'model': 'yatm'},
+        {'name': 'ocean', 'model': 'mom'}
+    ]
+    test_config['restart_freq'] = restart_freq
+    if restart_history:
+        test_config['restart_history'] = restart_history
+
+    write_config(test_config)
+
+
+@pytest.mark.parametrize(
+    "restart_freq, restart_history, expected_pruned_restarts_indices",
+    [
+        ("1MS", None, []),
+        ("2MS", None, [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23]),
+        ("2MS", 5, [1, 3, 5, 7, 9, 11, 13, 15, 17, 19]),
+        ("12MS", None,
+         [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11,
+          13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23]),
+        ("1YS", None,
+         [1, 2, 3, 4, 5, 6, 7, 8, 9, 10,
+          12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22]),
+        (1, 1, []),
+        (5, 5, [1, 2, 3, 4, 6, 7, 8, 9, 11, 12, 13, 14, 16, 17, 18, 19]),
+        (5, None, [1, 2, 3, 4, 6, 7, 8, 9, 11, 12, 13, 14, 16, 17, 18, 19])
```

Comment:
I guess I misunderstood, with `restart_history: None` I'd expect it to prune `21` as well. So by default all intermediate files are retained until a restart is pruned? This is good behaviour by default, so I'm not suggesting changing it, just wanted to confirm this.

In any case it would be prudent, and pretty easy, to include a case where `restart_history` is < `restart_freq` and vice_versa for the integer pruning options. Maybe make one that is a weird number, like 3 or 7?

Reply:
Yes, the default behaviour if there's no `restart_history` is to keep intermediate restarts. So in this case the permanently archived restarts are 0, 5, 10, 15, 20, and intermediate restarts are 21, 22, 23, 24. The intermediate restarts will be deleted once there's restart 25. 

Thanks for the suggestion about the more tests- I've added in those cases for `restart_history` being 3 and 7